builder/proxmox/common/config.go

```diff
@@ -62,11 +62,15 @@ type Config struct {
 	CloudInitStoragePool string `mapstructure:"cloud_init_storage_pool"`
 
 	AdditionalISOFiles []additionalISOsConfig `mapstructure:"additional_iso_files"`
-	VMInterface        string                 `mapstructure:"vm_interface"`
+	VMInterface        interfaceConfig        `mapstructure:"vm_interface"`
```

Comment:
Just a suggestion: For this one, could it be simpler to keep it as a string and have an internal `vmInterface netaddr.IP` ? 

Validation validates things, then we can use the IP everywhere. 

Another option could be to create a new packer.IP type that is a string, but that can return a `netaddr.IP`.

Reply:
I'm not sure what you mean by that. Shouldn't we have an option to select whether or not the IPv6 or v4 of an interface should be used. I thought it should resemble the HTTPIP config as much as possible since they both do the same thing.

Reply:
Ah, sorry, I should have asked first : could the fact that the interface is `ipv4` or `ipv6` detected from the `VMInterface` string ?

Reply:
ah, nope, yes you are right, sorry I misunderstood !